packages/analytics/src/utils.ts

```diff
@@ -40,3 +40,46 @@ export const encodeDataToQueryString = <T extends Event>(
 
     return params.toString();
 };
+
+const reportEventError = (
+    type: ReportEventProps['type'],
+    retry: ReportEventProps['retry'],
+    err: any,
+) => {
+    let errorMessage = err?.error?.message || err?.message;
+
+    if (typeof errorMessage !== 'string') {
+        // this should never happen
+        errorMessage = 'Unknown error.';
+    }
+
+    // to circumvent sentry inbound filter
+    if (errorMessage.includes('Failed to fetch')) {
+        errorMessage = 'Failed to analytics fetch.';
+    }
+
+    const reportedMessage = `Analytics report failed. Reporting '${type}' ${
+        retry ? 'again' : 'was unsuccessful'
+    }. ${errorMessage}`;
+
+    console.error(reportedMessage);
+};
+
+interface ReportEventProps {
+    type: Event['type'];
+    url: string;
+    options: RequestInit;
+    retry: boolean;
+}
+
+export const reportEvent = async ({ type, url, options, retry }: ReportEventProps) => {
+    try {
+        await fetch(url, options);
+    } catch (err) {
+        reportEventError(type, retry, err);
+
+        if (retry) {
+            reportEvent({ type, url, options, retry: false });
+        }
```

Comment:
I was thinking we can put a `setTimeout` here, let me know what you think

Reply:
I was thinking about it. 
Probably it does not help with a fetch timeout. However, it can help if user is offline.

What delay do you suggest?

Reply:
Added 1s